packages/remark-mdx/extract-imports-and-exports.js

```diff
@@ -0,0 +1,62 @@
+const {transformSync} = require('@babel/core')
+const declare = require('@babel/helper-plugin-utils').declare
+
+class BabelPluginExtractImportsAndExports {
+  constructor() {
+    const nodes = []
+    this.state = {nodes}
+
+    this.plugin = declare(api => {
+      api.assertVersion(7)
+
+      return {
+        visitor: {
+          ExportDefaultDeclaration(path) {
+            const {start} = path.node
+            nodes.push({type: 'export', start, default: true})
+          },
+          ExportNamedDeclaration(path) {
+            const {start} = path.node
+            nodes.push({type: 'export', start})
+          },
+          ImportDeclaration(path) {
+            const {start} = path.node
+            nodes.push({type: 'import', start})
+          }
+        }
+      }
+    })
+  }
+}
+
+const partitionString = (str, indices) => {
```

Comment:
This function is a bit complex (maybe because of the `.reduce`?) But from glossing over the file I’m not really sure what it does exactly. Maybe a different name, a comment, or a `.map`?

Reply:
This isn’t blocking btw, but just thought I’d mention it